codegen/src/main/java/org/web3j/codegen/SolidityFunctionWrapper.java

```diff
@@ -304,105 +348,154 @@ private String buildEventDefinitionName(String eventName) {
         return fields;
     }
 
-    private static MethodSpec buildConstructor(Class authType, String authName) {
-        return MethodSpec.constructorBuilder()
+    private static MethodSpec buildConstructor(Class authType, String authName,
+                                               boolean withGasProvider) {
+        MethodSpec.Builder toReturn = MethodSpec.constructorBuilder()
                 .addModifiers(Modifier.PROTECTED)
                 .addParameter(String.class, CONTRACT_ADDRESS)
                 .addParameter(Web3j.class, WEB3J)
-                .addParameter(authType, authName)
-                .addParameter(BigInteger.class, GAS_PRICE)
-                .addParameter(BigInteger.class, GAS_LIMIT)
-                .addStatement("super($N, $N, $N, $N, $N, $N)",
-                        BINARY, CONTRACT_ADDRESS, WEB3J, authName, GAS_PRICE, GAS_LIMIT)
-                .build();
+                .addParameter(authType, authName);
+
+        if (withGasProvider) {
```

Comment:
I think we should enforce the use of ContractGasProvider from web3j >= v.4. I.e. remove the deprecated code generation methods.